File: python/first_unique_character.py
from collections import defaultdict

class Solution:
    def firstUniqChar(s):
        count = defaultdict(int)

        for c in s:
            count[c] += 1

        for i, c in enumerate(s):
            if count[c] == 1:
                return i
        
        return -1
    
print(Solution.firstUniqChar("leetcode"))
print(Solution.firstUniqChar("loveleetcode"))
print(Solution.firstUniqChar("aabb"))

File: leet/first_unique_character.rb
def first_uniq_char(s)
    output = -1
    chars = s.chars.tally
    chars.each do |k, v|
        if v == 1
            output = s.index(k)
            return output
        end
    end
    
    return output
end

puts first_uniq_char("leetcode")
puts first_uniq_char("loveleetcode")
puts first_uniq_char("aabb")
